src/openms/source/MATH/STATISTICS/KernelDensityEstimation.cpp

```diff
@@ -0,0 +1,310 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Justin Sing $
+// $Authors: Justin Sing $
+// --------------------------------------------------------------------------
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+// Use the evergreen umbrella header so the vendor headers are included
+// consistently inside the `evergreen` namespace and their helper
+// macros are defined. Then include the FFT submodule *inside* the
+// `evergreen` namespace so its template and type definitions live in
+// `evergreen::` as the rest of the code expects.
+#include <Evergreen/evergreen.hpp>
+namespace evergreen {
+#include <FFT/FFT.hpp>
+}
+
+#include <OpenMS/MATH/MISC/CubicSpline2d.h>
+#include <OpenMS/MATH/STATISTICS/KernelDensityEstimation.h>
+#include <OpenMS/MATH/StatisticFunctions.h>
+
+namespace OpenMS
+{
+  namespace Math
+  {
+
+    // Helper: next power of two >= v
```

Comment:
maybe just 
static std::size_t next_pow2(std::size_t v)
{
  return std::bit_ceil(v ? v : std::size_t{1});
}

Reply:
maybe then you could even remove method?